agent is restarting when pod is not found

https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/66864979093c63a6dc423f435633b553a77d4e40/metrics/metadata.go#L52

this error is not handled gracefully.
We checked one pod ids and it was a CI runner that was done with the job.

```
level=error msg="Metadata: Pod not found" podUid=2fd21870-xxxx-xxxx-xxxx-4eea5b905015
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xa89ce9]
```

--- a/metrics/accumulator.go
+++ b/metrics/accumulator.go
@@ -56,7 +56,13 @@ func (a *MetricDataAccumulator) podStats(podResource *Resource, s stats.PodStats
 		"name": podResource.Name,
 	}).Trace("podStats")
 
-	if !a.metricGroupsToCollect[PodMetricGroup] || podResource.PodMetadata.Pod == nil {
+	if !a.metricGroupsToCollect[PodMetricGroup] {
+		return
+	}
+	
+	// Metatdata can be nil if pod is terminted before metodata is fetched.
+	// no metrics are needed.
+	if podResource.PodMetadata == nil {
 		return
 	}
 

--- a/metrics/accumulator.go
+++ b/metrics/accumulator.go
@@ -60,7 +60,7 @@ func (a *MetricDataAccumulator) podStats(podResource *Resource, s stats.PodStats
 		return
 	}
 	
-	// Metatdata can be nil if pod is terminted before metodata is fetched.
+	// Metatdata can be nil if pod is terminated before metadata is fetched.
 	// no metrics are needed.
 	if podResource.PodMetadata == nil {
 		return

--- a/metrics/accumulator.go
+++ b/metrics/accumulator.go
@@ -56,7 +56,7 @@ func (a *MetricDataAccumulator) podStats(podResource *Resource, s stats.PodStats
 		"name": podResource.Name,
 	}).Trace("podStats")
 
-	if !a.metricGroupsToCollect[PodMetricGroup] {
+	if !a.metricGroupsToCollect[PodMetricGroup] || podResource.PodMetadata.Pod == nil {
 		return
 	}
 

--- a/metrics/metadata.go
+++ b/metrics/metadata.go
@@ -50,7 +50,7 @@ func (m *Metadata) GetPodMetadataByUid(uid types.UID) (*PodMetadata, error) {
 	logrus.WithFields(logrus.Fields{
 		"podUid": uid,
 	}).Error("Metadata: Pod not found")
-	return &PodMetadata{}, errors.New("pod not found")
+	return nil, errors.New("pod not found")
 }
 
 type NodeMetadata struct {

--- a/metrics/metrics_test.go
+++ b/metrics/metrics_test.go
@@ -360,6 +360,20 @@ func TestNodeStats(t *testing.T) {
 	assert.Equal(t, 1, len(data.Resource.Labels))
 }
 
+func TestPodStatsOnDeadPod(t *testing.T) {
+	summary, metadata := createMockSourceAssets(true, true, nil, true)
+	acc := createMockAccumulator(metadata, ValidMetricGroups)
+
+	podStats := getPodStatsByUID(summary.Pods, "5997ad9b-1d2a-43cf-ab57-a98d8796dc34")
+	node := getNodeResource(summary.Node, metadata)
+	pod := getPodResource(node, podStats, metadata)
+	pod.PodMetadata = nil
+
+	acc.podStats(pod, podStats)
+	assert.Equal(t, 0, len(acc.Data))
+
+}
+
 func TestPodStats(t *testing.T) {
 	summary, metadata := createMockSourceAssets(true, true, nil, true)
 	acc := createMockAccumulator(metadata, ValidMetricGroups)

